### PR TITLE
Incorrect shutdown screen

### PR DESCRIPTION
**Pull request by VladPavelescu**
[Jira ticket](https://yetitoolsmartsoftware.atlassian.net/browse/YTSS-1992?atlOrigin=eyJpIjoiMTc2YjE2NTE4ZTg4NDFiZWE3Zjc2MDNiOTgzZDQ3OWIiLCJwIjoiaiJ9)

Tested on the rig and on big console. 

A "correct_shutdown" flag gets set to False if it doesn't already exist or if it is currently set to True on startup. The flag gets set to true only from console_utils when the shutdown or reboot functions are called.

image

### CODE REVIEW

**Review by LettyYeti — CHANGES_REQUESTED** [review]
- Add new screen to start_up_sequence_test, or write appropriate visual screen test for it
- Need to resize top banner (see video in slack)
- Need to find other valid places that the console is shutdown and rebooted, at the moment the config is only updated if user shuts down console from the lobby and there are many other places where it can be shut down or rebooted :)

**Comment by Masc999** [comment]
I would even go further:
have one class that does the actual sudo calls for shutdown, restart and cancel. This function sets the flag to True.
find all instances where the console is reboot or shutdown and call the implemented functions instead.

**Comment by Masc999** [comment]
> I would even go further: have one class that does the actual sudo calls for shutdown, restart and cancel. This function sets the flag to True. find all instances where the console is reboot or shutdown and call the implemented functions instead.

File: src/asmcnc/skavaUI/popup_info.py [hunk 1]
```diff
@@ -1107,6 +1107,7 @@ def cancel_shutdown(*args):
             os.system('sudo shutdown -c')
 
         def shutdown_now(*args):
+            os.system('sudo sed -i "s/correct_shutdown=False/correct_shutdown=True/" config.txt')
```
Comment (Masc999) [inline src/asmcnc/skavaUI/popup_info.py:1110]: have one class that does the actual sudo calls for shutdown, restart and cancel. This function sets the flag to True.
find all instances where the console is reboot or shutdown and call the implemented functions instead.

File: src/asmcnc/apps/start_up_sequence/start_up_sequence_manager.py [hunk 2]
```diff
@@ -123,6 +127,16 @@ def show_user_pro_plus_safety(self):
 		if ('False' in pro_plus_safety or not pro_plus_safety) and os.path.exists(self.m.theateam_path): return True
 		else: return False
 
+	def show_incorrect_shutdown_screen(self):
+		correct_shutdown = (os.popen('grep "correct_shutdown" config.txt').read())
+		if ('False' in correct_shutdown): return True
+		else:
+			if not correct_shutdown:
```
Comment (Masc999) [inline src/asmcnc/apps/start_up_sequence/start_up_sequence_manager.py:134]: have one class that does the actual sudo calls for shutdown, restart and cancel. This function sets the flag to True.
find all instances where the console is reboot or shutdown and call the implemented functions instead.

**Review by Masc999 — CHANGES_REQUESTED** [review]

File: src/asmcnc/apps/start_up_sequence/start_up_sequence_manager.py (hunk 2 above)
Comment (Masc999) [reply src/asmcnc/apps/start_up_sequence/start_up_sequence_manager.py:134]: All the file handling and flag re-setting should be done in a, let's call it, ShutdownHandler. It should also have the shutdown functions from console_utils. The start_up_sequence_manager looks the correct_shutdown state up in the ShutdownHandler.

File: src/asmcnc/production/lower_beam_qc_jig/lower_beam_qc.py [hunk 3]
```diff
@@ -330,8 +331,7 @@ def enable_alarms(self):
 		self.m.s.write_command('$21 = 1')
 
 	def shutdown_console(self):
```
Comment (Masc999) [inline src/asmcnc/production/lower_beam_qc_jig/lower_beam_qc.py:333]: Why have this function in the first place? just call console_utils.shutdown() directly

File: src/asmcnc/production/lower_beam_qc_jig/lower_beam_qc_warranty.py [hunk 4]
```diff
@@ -285,8 +286,7 @@ def enable_alarms(self):
 		self.m.s.write_command('$21 = 1')
 
 	def shutdown_console(self):
```
Comment (Masc999) [inline src/asmcnc/production/lower_beam_qc_jig/lower_beam_qc_warranty.py:288]: Why have this function in the first place? just call console_utils.shutdown() directly

File: src/asmcnc/production/lowerbeam_calibration_jig/lowerbeam_calibration_success.py [hunk 5]
```diff
@@ -62,8 +63,7 @@ def enter_prev_screen(self):
         self.sm.current = 'lbc4'
 
     def shutdown_console(self):
```
Comment (Masc999) [inline src/asmcnc/production/lowerbeam_calibration_jig/lowerbeam_calibration_success.py:65]: Why have this function in the first place? just call console_utils.shutdown() directly

File: src/asmcnc/production/z_head_qc_jig/z_head_qc_aftr_apr_21.py [hunk 6]
```diff
@@ -786,8 +787,7 @@ def update_status_text(self, dt):
         self.consoleStatusText.text = self.sm.get_screen('home').gcode_monitor_widget.consoleStatusText.text
 
     def do_reboot(self):
```
Comment (Masc999) [inline src/asmcnc/production/z_head_qc_jig/z_head_qc_aftr_apr_21.py:789]: Why have this function in the first place? just call console_utils.reboot() directly

File: src/asmcnc/production/z_head_qc_jig/z_head_qc_b4_apr_21.py [hunk 7]
```diff
@@ -585,8 +586,7 @@ def update_status_text(self, dt):
         self.consoleStatusText.text = self.sm.get_screen('home').gcode_monitor_widget.consoleStatusText.text
 
     def do_reboot(self):
```
Comment (Masc999) [inline src/asmcnc/production/z_head_qc_jig/z_head_qc_b4_apr_21.py:588]: Why have this function in the first place? just call console_utils.reboot() directly

File: src/asmcnc/apps/systemTools_app/screens/screen_factory_settings.py [hunk 8]
```diff
@@ -911,7 +913,7 @@ def close_sw(self, dt):
         sys.exit()
 
     def shutdown_console(self, dt):
```
Comment (Masc999) [inline src/asmcnc/apps/systemTools_app/screens/screen_factory_settings.py:915]: Why have this function in the first place? just call console_utils.shutdown() directly

File: src/asmcnc/production/z_head_qc_jig/z_head_qc_home.py [hunk 9]
```diff
@@ -182,6 +183,5 @@ def secret_option_c(self):
         self.sm.current = 'qcWC'
 
     def shutdown_console(self):
```
Comment (Masc999) [inline src/asmcnc/production/z_head_qc_jig/z_head_qc_home.py:185]: Why have this function in the first place? just call console_utils.shutdown() directly

File: src/asmcnc/skavaUI/popup_info.py [hunk 10]
```diff
@@ -1104,10 +1105,10 @@ def __init__(self, screen_manager, localization):
         cancel_string = self.l.get_bold('Cancel')
 
         def cancel_shutdown(*args):
```
Comment (Masc999) [inline src/asmcnc/skavaUI/popup_info.py:1107]: Why have this function in the first place? just call console_utils directly

File: src/asmcnc/skavaUI/popup_info.py [hunk 11]
```diff
@@ -1104,10 +1105,10 @@ def __init__(self, screen_manager, localization):
         cancel_string = self.l.get_bold('Cancel')
 
         def cancel_shutdown(*args):
-            os.system('sudo shutdown -c')
+            console_utils.cancel_shutdown()
 
         def shutdown_now(*args):
```
Comment (Masc999) [inline src/asmcnc/skavaUI/popup_info.py:1110]: Why have this function in the first place? just call console_utils directly

File: src/asmcnc/skavaUI/screen_rebooting.py [hunk 12]
```diff
@@ -64,5 +65,4 @@ def on_enter(self):
         Clock.schedule_once(self.reboot, 1)
 
     def reboot(self, dt):
```
Comment (Masc999) [inline src/asmcnc/skavaUI/screen_rebooting.py:67]: Why have this function in the first place? just call console_utils directly

**Review by Masc999 — CHANGES_REQUESTED** [review]

File: src/asmcnc/apps/start_up_sequence/start_up_sequence_manager.py (hunk 2 above)
Comment (Masc999) [reply src/asmcnc/apps/start_up_sequence/start_up_sequence_manager.py:134]: Now I know again why I wrote this: start_up_sequence_manager.py is reading the file as well. It should just read the flag from console_utils like 
`if console_utils.correct_shutdown() : do_something...`
Please fix that, so the file handling is encapsulated in console_utils

File: src/asmcnc/production/z_head_qc_jig/z_head_qc_b4_apr_21.py (hunk 7 above)
Comment (Masc999) [reply src/asmcnc/production/z_head_qc_jig/z_head_qc_b4_apr_21.py:586]: please call console_utils directly

File: src/asmcnc/production/z_head_qc_jig/z_head_qc_aftr_apr_21.py (hunk 6 above)
Comment (Masc999) [reply src/asmcnc/production/z_head_qc_jig/z_head_qc_aftr_apr_21.py:788]: Please call console_utils directl<

**Pull request merged**